src/utils/survey-event-receiver.ts

```diff
@@ -0,0 +1,60 @@
+import { Survey } from '../posthog-surveys-types'
+
+export class SurveyEventReceiver {
+    private eventRegistry: Map<string, string[]>
+    private surveysActivatedKey = 'surveysActivated'
+
+    constructor() {
+        this.eventRegistry = new Map<string, string[]>()
+    }
+
+    register(surveys: Survey[]): void {
+        surveys.forEach((survey) => {
+            if (survey.events && survey.events.length > 0) {
+                this.eventRegistry.set(survey.id, survey.events)
+            }
+        })
+    }
+
+    deRegister(survey: Survey): void {
+        if (this.eventRegistry.has(survey.id)) {
+            this.eventRegistry.delete(survey.id)
+        }
+    }
+
+    on(event: string): void {
+        const activatedSurveys: string[] = []
+        this.eventRegistry.forEach((events, surveyID) => {
+            if (events.includes(event)) {
+                activatedSurveys.push(surveyID)
+            }
+        })
+
+        let surveys: string[] = []
+        if (sessionStorage.getItem(this.surveysActivatedKey)) {
+            surveys = JSON.parse(<string>sessionStorage.getItem(this.surveysActivatedKey)) as unknown as string[]
+        }
+        surveys.concat(activatedSurveys)
+        sessionStorage.setItem(this.surveysActivatedKey, JSON.stringify(surveys))
+    }
```

Comment:
IMO, this could be written in a slightly more compiler-friendly way, like this (saves you needing to do as much type casting with the `as` keyword, which can be dangerous).  I think this pattern (of assigning the result of the localStorage responses to a variable and then branching of that variable to appease the compiler) applies to several of these class methods, but I just gave one example here.

```suggestion
    on(event: string): void {
        const activatedSurveys: string[] = []

        this.eventRegistry.forEach((events, surveyID) => {
            if (events.includes(event)) {
                activatedSurveys.push(surveyID)
            }
        })

        const existingActivatedSurveys = sessionStorage.getItem(this.surveysActivatedKey)
        const existingSurveys: string[] = existingActivatedSurveys ? JSON.parse(existingActivatedSurveys) : []

        const updatedSurveys = existingSurveys.concat(activatedSurveys)
        sessionStorage.setItem(this.surveysActivatedKey, JSON.stringify(updatedSurveys))
    }
```